### PR TITLE
IGNITE-15183 : Add thin clients thread pool monitoring

### PR DESCRIPTION
**Pull request by Vladsz83**
Changes: thin client pool has been moved into PoolProcessor

### CODE REVIEW

File: modules/core/src/main/java/org/apache/ignite/internal/processors/pool/PoolProcessor.java [hunk 1]
```diff
@@ -498,6 +503,18 @@ public PoolProcessor(GridKernalContext ctx) {
...
+        if (cfg.getClientConnectorConfiguration() != null) {
+            thinClientExec = new IgniteThreadPoolExecutor(
+                "client-connector",
+                cfg.getIgniteInstanceName(),
+                cfg.getClientConnectorConfiguration().getThreadPoolSize(),
+                cfg.getClientConnectorConfiguration().getThreadPoolSize(),
+                0,
+                new LinkedBlockingQueue<>(),
+                GridIoPolicy.UNDEFINED,
+                new OomExceptionHandler(ctx));
```
Comment (alex-plekhanov) [inline modules/core/src/main/java/org/apache/ignite/internal/processors/pool/PoolProcessor.java:515]: `new OomExceptionHandler(ctx)` -> `oomeHnd`?

File: modules/core/src/main/java/org/apache/ignite/internal/processors/odbc/ClientListenerProcessor.java [hunk 2]
```diff
@@ -366,12 +354,6 @@ private void unregisterMBean() {
 
             ctx.ports().deregisterPorts(getClass());
 
-            if (execSvc != null) {
-                U.shutdownNow(getClass(), execSvc, log);
-
-                execSvc = null;
```
Comment (alex-plekhanov) [inline modules/core/src/main/java/org/apache/ignite/internal/processors/odbc/ClientListenerProcessor.java:372]: Let's keep only this line

File: modules/core/src/main/java/org/apache/ignite/internal/processors/pool/PoolProcessor.java (hunk 1 above)
Comment (Vladsz83) [reply modules/core/src/main/java/org/apache/ignite/internal/processors/pool/PoolProcessor.java]: Fixed

File: modules/core/src/main/java/org/apache/ignite/internal/processors/odbc/ClientListenerProcessor.java (hunk 2 above)
Comment (Vladsz83) [reply modules/core/src/main/java/org/apache/ignite/internal/processors/odbc/ClientListenerProcessor.java:372]: Fixed

**Pull request closed without merging**